pkg/operator/vsphere_check.go

```diff
@@ -52,26 +51,32 @@ func (v *vSphereChecker) runChecks(ctx context.Context, clusterInfo *util.Cluste
 	}
 
 	defer func() {
-		if err := checkContext.GovmomiClient.Logout(ctx); err != nil {
-			klog.Errorf("Failed to logout: %v", err)
+		for _, vCenter := range checkContext.VCenters {
+			if err := vCenter.GovmomiClient.Logout(ctx); err != nil {
+				klog.Errorf("Failed to logout: %v", err)
+			}
 		}
 	}()
 
 	// Get the fully-qualified vsphere username
-	sessionMgr := session.NewManager(checkContext.VMClient)
-	user, err := sessionMgr.UserSession(ctx)
-	if err != nil {
-		return resultCollector, err
+	for index := range checkContext.VCenters {
+		vc := checkContext.VCenters[index]
+		sessionMgr := session.NewManager(vc.VMClient)
+		user, err := sessionMgr.UserSession(ctx)
+		if err != nil {
+			return resultCollector, err
+		}
+
+		vc.AuthManager = object.NewAuthorizationManager(vc.VMClient)
+		vc.Username = user.UserName
```

Comment:
`vc.AuthManager` and `vc.Username` will be visible outside this function even if `v.controller.GetInfrastructure(ctx)` fails below. This was not the case before this PR. Not sure if it's an issue, but the way how `runChecks()` was implemented suggests it could matter (otherwise, why it used local vars for `sessionMgr` and `authManager`? it could set it inline).

Reply:
I can set the user/pass after the infra check.  the check instance was already setting the user/pass as instance variables, i have just moved them into a vCenter object since we need multiple.  I'll look into moving setting of user/pass to after infra to maintain current logical flow.